underfs/oss/src/main/java/alluxio/underfs/oss/StsOssClientProvider.java

```diff
@@ -0,0 +1,239 @@
+
+  private UnderFileSystemConfiguration mOssConf;
+  private final ScheduledExecutorService mRefreshOssClientScheduledThread;
+
+  /**
+   * Constructs a new instance of {@link StsOssClientProvider}.
+   * @param ossConfiguration {@link UnderFileSystemConfiguration} for OSS
+   * @throws IOException if failed to init OSS STS client
+   */
+  public StsOssClientProvider(UnderFileSystemConfiguration ossConfiguration) throws IOException {
+    mEcsMetadataService = ossConfiguration.getString(
+        PropertyKey.UNDERFS_OSS_STS_ECS_METADATA_SERVICE_ENDPOINT);
+    mTokenTimeoutMs = ossConfiguration.getMs(PropertyKey.UNDERFS_OSS_STS_TOKEN_TIMEOUT_MS);
+    RetryPolicy retryPolicy = new ExponentialBackoffRetry(
+        BASE_SLEEP_TIME_MS, MAX_SLEEP_MS, MAX_RETRIES);
+    IOException lastException = null;
+    while (retryPolicy.attempt()) {
+      try {
+        initializeOssClient(ossConfiguration);
+        lastException = null;
+        break;
+      } catch (IOException e) {
+        LOG.warn("init oss client failed! has retry {} times", retryPolicy.getAttemptCount(), e);
+        lastException = e;
+      }
+    }
+    if (lastException != null) {
+      throw lastException;
+    }
+    mRefreshOssClientScheduledThread = Executors.newSingleThreadScheduledExecutor(
+        ThreadFactoryUtils.build("refresh_oss_client-%d", false));
+    mRefreshOssClientScheduledThread.scheduleAtFixedRate(() -> {
+      try {
+        if (null != mOssConf) {
+          refreshOssStsClient(mOssConf);
+        }
+      } catch (Exception e) {
+        //retry it
+        LOG.warn("throw exception when clear meta data cache", e);
+      }
+    }, 0, 60000, TimeUnit.MILLISECONDS);
+  }
+
+  protected void refreshOssStsClient(UnderFileSystemConfiguration ossConfiguration)
+      throws IOException {
+    ClientBuilderConfiguration ossClientConf = getClientBuilderConfiguration(ossConfiguration);
+    createOrRefreshStsOssClient(ossConfiguration, ossClientConf);
+  }
+
+  private ClientBuilderConfiguration getClientBuilderConfiguration(
+      UnderFileSystemConfiguration ossConfiguration) {
+    ClientBuilderConfiguration ossClientConf = new ClientBuilderConfiguration();
+    ossClientConf.setMaxConnections(ossConfiguration.getInt(PropertyKey.UNDERFS_OSS_CONNECT_MAX));
+    ossClientConf.setMaxErrorRetry(ossConfiguration.getInt(PropertyKey.UNDERFS_OSS_RETRY_MAX));
+    ossClientConf.setConnectionTimeout(
+        (int) ossConfiguration.getMs(PropertyKey.UNDERFS_OSS_CONNECT_TIMEOUT));
+    ossClientConf.setSocketTimeout(
+        (int) ossConfiguration.getMs(PropertyKey.UNDERFS_OSS_SOCKET_TIMEOUT));
+    ossClientConf.setSupportCname(false);
+    ossClientConf.setCrcCheckEnabled(true);
+    return ossClientConf;
+  }
+
+  /**
+   * Init the STS OSS client.
+   * @param ossConfiguration OSS {@link UnderFileSystemConfiguration}
+   * @throws IOException if failed to init OSS client
+   */
+  public void initializeOssClient(UnderFileSystemConfiguration ossConfiguration)
```

Comment:
why `public` if it's only used in this class? If this is intended to called from outside, is the implementation thread safe?

Reply:
fixed